src/main/java/org/terasology/module/behaviors/actions/FindPathToNode.java

```diff
@@ -2,95 +2,84 @@
 // SPDX-License-Identifier: Apache-2.0
 package org.terasology.module.behaviors.actions;
 
-import com.google.common.util.concurrent.FutureCallback;
-import com.google.common.util.concurrent.Futures;
-import com.google.common.util.concurrent.MoreExecutors;
-import com.google.common.util.concurrent.SettableFuture;
 import org.joml.Vector3f;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
 import org.terasology.engine.logic.behavior.BehaviorAction;
 import org.terasology.engine.logic.behavior.core.Actor;
 import org.terasology.engine.logic.behavior.core.BaseAction;
 import org.terasology.engine.logic.behavior.core.BehaviorState;
+import org.terasology.engine.logic.location.LocationComponent;
 import org.terasology.engine.registry.CoreRegistry;
 import org.terasology.engine.registry.In;
+import org.terasology.engine.world.block.Blocks;
+import org.terasology.flexiblepathfinding.JPSConfig;
+import org.terasology.flexiblepathfinding.PathfinderCallback;
+import org.terasology.flexiblepathfinding.PathfinderSystem;
 import org.terasology.module.behaviors.components.MinionMoveComponent;
-import org.terasology.navgraph.NavGraphSystem;
-import org.terasology.navgraph.WalkableBlock;
-import org.terasology.pathfinding.componentSystem.PathfinderSystem;
-import org.terasology.pathfinding.model.Path;
+import org.terasology.module.behaviors.systems.PluginSystem;
 
-import java.util.Arrays;
 import java.util.List;
 
 /**
- * Requests a path to a target defined using the <b>MinionMoveComponent.target</b>.<br/> <br/>
- * <b>SUCCESS</b> / <b>FAILURE</b>: when paths is found or not found (invalid).<br/>
- * <b>RUNNING</b>: as long as path is searched.<br/>
- * <br/> Auto generated javadoc - modify README.markdown instead!
+ * Finds a path to the pathGoalPosition of the Actor, stores it in FlexibileMovementComponent.path
+ * <p/>
+ * SUCCESS: When the pathfinder returns a valid path
+ * <p/>
+ * FAILURE: When the pathfinder returns a failure or invalid path
+ *
  */
 @BehaviorAction(name = "find_path")
 public class FindPathToNode extends BaseAction {
 
     @In
-    private transient NavGraphSystem navGraphSystem;
+    PathfinderSystem pathfinderSystem;
```

Comment:
@DarkWeird

Reply:
Hmmm. 

Non private - 
      1. prepare to gestalt-di --- though stuck somewhere in my mind.
      2. holywar about should be injectable(autowirable in spring) fields as private
non transient - it mistake. it can produce error when you try save behavio from behavior editor.

Reply:
```suggestion
    transient PathfinderSystem pathfinderSystem;
```